### PR TITLE
Fix `compile_like_murdock.py` when board is empty

### PR DESCRIPTION
**Pull request by MrKevinWeiss**
### Contribution description

Turns out we didn't test the simplest case, board being empty.

I added some automatic testing to check that and we can build when needed...

### Testing procedure

Green `tools-test` action

### CODE REVIEW

File: dist/tools/compile_test/tests/test.sh [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+        echo -e "${GREEN}✓$NO_COLOUR"
+    else
+        echo -e "${RED}x$NO_COLOUR"
+    fi
+}
+
+
+# sets the global accumulative RESULT variable to keep record of errors during
+# tests.
+function set_result {
```
Comment (Teufelchen1) [inline dist/tools/compile_test/tests/test.sh:23]: Try naming the function after its purpose, not what it actually does. e.g. "accumulate_errors".

File: dist/tools/compile_test/tests/test.sh [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+    echo -n "Running test: \"$*\" "
+    $1
+    res=$?
+
+    print_result $res
+    set_result $res
+}
+
+function call_with_defaults {
+    if  OUT=$(${COMPILE_TEST_DIR}/compile_like_murdock.py -d 2>&1)
```
Comment (Teufelchen1) [inline dist/tools/compile_test/tests/test.sh:44]: use `--dry-run` to enhance the readability for (almost) zero costs!

File: dist/tools/compile_test/tests/test.sh [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+    then
+        return 0
+    else
+        1>&2 echo -n "$OUT"
```
Comment (Teufelchen1) [inline dist/tools/compile_test/tests/test.sh:48]: Couldn't this be moved to `run_test`?

**Review by Teufelchen1 — CHANGES_REQUESTED** [review]
Nice!
I found some nit's and please take a look at the static checks.
Worked like a charm!  (you missed an opportunity to use ✅ / ❌ though ;-))

File: dist/tools/compile_test/tests/test.sh (hunk 2 above)
Comment (Teufelchen1) [reply dist/tools/compile_test/tests/test.sh:44]: Ah, ofc, this also works with `--apps` and `--boards`

File: dist/tools/compile_test/tests/test.sh [hunk 4]
```diff
@@ -0,0 +1,67 @@
+#!/usr/bin/env bash
+
```
[the remark this answers is not in the document]
Comment (MrKevinWeiss) [reply dist/tools/compile_test/tests/test.sh:6]: Done

File: dist/tools/compile_test/tests/test.sh (hunk 1 above)
Comment (MrKevinWeiss) [reply dist/tools/compile_test/tests/test.sh]: Done

File: dist/tools/compile_test/tests/test.sh (hunk 2 above)
Comment (MrKevinWeiss) [reply dist/tools/compile_test/tests/test.sh]: Done

File: dist/tools/compile_test/tests/test.sh (hunk 3 above)
Comment (MrKevinWeiss) [reply dist/tools/compile_test/tests/test.sh]: ChatGPT helped with a bit of refactoring.

**Comment by MrKevinWeiss** [comment]
@Teufelchen1 have I addressed all your commented? May I squash?

**Comment by Teufelchen1** [comment]
Hi! Yes squash & merge. Thanks for waiting!

**Comment by MrKevinWeiss** [comment]
Done, I will need an ack to merge :point_right::point_left:

**Comment by MrKevinWeiss** [comment]
bors merge

**Comment by MrKevinWeiss** [comment]
bors cancel

**Comment by MrKevinWeiss** [comment]
bors merge

**Pull request merged**